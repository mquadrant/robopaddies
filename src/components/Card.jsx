import React from 'react'

export default function Card({name,email}) {
    return (
        <div className="card flex-none inline-block bg-green-200 shadow-md p-3 m-2 rounded-lg">
            <img src={`https://robohash.org/${name}?size=200x200`} alt={name}/>
            <div className="my-2">
                <h4 className="font-semibold text-base text-sm text-center">{name}</h4>
                <p className="text-center text-xs">{email}</p>
            </div>
        </div>
    )
}
