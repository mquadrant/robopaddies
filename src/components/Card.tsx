import React from 'react'

interface ICard {
    name: string;
    email: string;
    key?:number;
    id?:number
}

export default function Card({name,email}:ICard) {
    return (
        <div className="card flex-none inline-block bg-green-200 shadow-md p-3 m-2 rounded-lg">
            <img src={`https://robohash.org/${name}?size=200x200`} alt={name}/>
            <div className="my-2">
                <h4 className="font-semibold text-base text-sm text-center">{name}</h4>
                <p className="text-center text-xs">{email}</p>
            </div>
        </div>
    )
}
