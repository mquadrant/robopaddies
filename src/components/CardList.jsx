import React from 'react'
import Card from './Card'

export default function CardList(props) {
    const {robots} = props;
    return (
        <div className="mt-8 container mx-auto flex flex-wrap justify-center">
            {robots?robots.map((robot,index)=>{
                return (<Card key={index} id= {robot.id} name={robot.name} email={robot.email}/>)
            }):null}
      </div>
    )
}
