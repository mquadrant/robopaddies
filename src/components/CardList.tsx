import React from 'react'
import Card from './Card'

export default function CardList(props:any) {
    const {robots} = props;
    return (
        <div className="mt-8 container mx-auto flex flex-wrap justify-center">
            {robots?robots.map((robot:any,index:any)=>{
                return (<Card key={index} id= {robot.id} name={robot.name} email={robot.email}/>)
            }):null}
      </div>
    )
}
