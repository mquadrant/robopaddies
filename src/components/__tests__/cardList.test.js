import React from 'react';
import { shallow,mount} from 'enzyme';
import CardList from './../CardList';
import Card from './../Card';

it('expect to render the Card component', () => {
    const wrapper = shallow(<CardList />);
    expect(wrapper.length).toEqual(1);
  });

it('expect to render Card component with snapshot', () => {
    const mockRobot = [
        {
            id:1,
            name:"John Doe",
            username:"Johnny",
            email:"John@gmail.com"
        }
    ]
    const wrapper = shallow(<CardList robots={mockRobot}/>)
    expect(wrapper.find(Card)).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('expect Card component to have robots prop', () => {
    const mockRobot = [
        {
            id:1,
            name:"John Doe",
            username:"Johnny",
            email:"John@gmail.com"
        }
    ]
    const wrapper = mount(<CardList robots={mockRobot}/>)
    expect(wrapper.props().robots).toEqual(mockRobot);
  });
