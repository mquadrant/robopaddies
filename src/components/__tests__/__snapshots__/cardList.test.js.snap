// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component with snapshot 1`] = `
<div
  className="mt-8 container mx-auto flex flex-wrap justify-center"
>
  <Card
    email="John@gmail.com"
    id={1}
    key="0"
    name="John Doe"
  />
</div>
`;
